heidrun/ct_dc.rb

```diff
@@ -0,0 +1,95 @@
+
+# If more than one dc:contributor property exists then it drops the last one
+# If one or zero exist return nil
+contributor_top = lambda do |r|
+  contrib = r['dc:contributor']
+  count = r['dc:contributor'].count
+  (count > 1) ? contrib.first_value(count - 1) : contrib.last_value(0)
+end
+
+Krikri::Mapper.define(:ct_dc,
+                      :parser => Krikri::OaiDcParser,
+                      :parser_args => '//rdf:RDF/rdf:Description') do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/connecticut'
+    label 'Connecticut Digital Library'
+  end
+
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('dc:contributor').last_value
+  end
+
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('dc:identifier')
+  end
+
+  preview :class => DPLA::MAP::WebResource do
+    uri record.field('dc:source')
+  end
+
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+    collection  :class => DPLA::MAP::Collection,
+                :each => record.field('xmlns:header', 'xmlns:setSpec'),
+                :as => :coll do
+      title coll
+    end
+
+    contributor :class => DPLA::MAP::Agent,
+                :each => record.map( &contributor_top ).flatten,
+                :as => :contributor_name do
+      providedLabel contributor_name
+    end
+
+    creator :class => DPLA::MAP::Agent,
+            :each => record.field('dc:creator'),
+            :as => :creator do
+      providedLabel creator
+    end
+
+    date  :class => DPLA::MAP::TimeSpan,
+          :each => record.field('dc:date'),
+          :as => :date do
+      providedLabel date
+    end
+
+    description record.field('dc:description')
+
+    dcformat record.fields('dc:format')
+
+    language  :class => DPLA::MAP::Controlled::Language,
+              :each => record.field('dc:language'),
+              :as => :lang do
+      prefLabel lang
+    end
+
+    spatial :class => DPLA::MAP::Place,
+            :each => record.field('dc:coverage'),
+            :as => :place do
+      providedLabel place
+    end
+
+    publisher :class => DPLA::MAP::Agent,
+    					:each => record.field('dc:publisher'),
```

Comment:
Still in favor of reformatting this, and probably reformatting the file in general. A rubocop run might be called for here.